plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/plugin.xml

```diff
@@ -8,25 +8,35 @@
   <extension point="org.eclipse.ui.propertyPages">
     <page
         id="com.google.cloud.tools.eclipse"
-        name="Google"
+        name="Google Cloud Platform"
         class="com.google.cloud.tools.eclipse.properties.GooglePropertyPage">
+       <enabledWhen>
+         <reference definitionId="com.google.cloud.tools.eclipse.appengine.onlyInGCPProjects" />
+       </enabledWhen>
     </page>
   </extension>
-
+  
   <extension point="org.eclipse.ui.propertyPages">
     <page
         id="com.google.cloud.tools.eclipse.appengine.deploy"
         name="App Engine Deployment"
         category="com.google.cloud.tools.eclipse"
         class="com.google.cloud.tools.eclipse.appengine.deploy.ui.DeployPropertyPage">
-      <enabledWhen> <!-- this will probably be too strict when App Engine Flex becomes supported -->
-        <adapt type="org.eclipse.core.resources.IProject">
-          <test
-              forcePluginActivation="true"
-              property="org.eclipse.wst.common.project.facet.core.projectFacet"
-              value="com.google.cloud.tools.eclipse.appengine.facet" />
-        </adapt>
+      <enabledWhen>
+        <reference definitionId="com.google.cloud.tools.eclipse.appengine.onlyInGCPProjects" />
       </enabledWhen>
     </page>
   </extension>
+
+  <extension point="org.eclipse.core.expressions.definitions">
+    <definition id="com.google.cloud.tools.eclipse.appengine.onlyInGCPProjects">
+       <!-- this will probably be too strict when App Engine Flex becomes supported -->
+      <adapt type="org.eclipse.core.resources.IProject">
+        <test
+            forcePluginActivation="true"
```

Comment:
`forcePluginActivation` should be avoided unless its known to be necessary.  `.wst.common.project.facet.core` should have been loaded if there is a faceted project to construct classpaths, etc.  So if it's not loaded then we almost certainly don't have a faceted project.